### ISSUE TITLE
npm build 25.1.0 (latest) missing build files

### ISSUE THREAD

**Issue by craftycram** [matrix-org/matrix-js-sdk#3363]
## Describe the bug
The npm-hosted build is missing all build files (`/lib` and `/dist` folder)

## To Reproduce
Just try installing the newest version of `matrix-js-sdk` using npm or yarn.  
npm: `npm install matrix-js-sdk` / `npm install matrix-js-sdk@25.1.0`  
yarn: `yarn add matrix-js-sdk` / `yarn add matrix-js-sdk@25.1.0`

## Expected behavior
I'd expect npm / yarn to install a build of the sdk containing the built javascript files as specified in the packages `package.json`:
```
...
"main": "./lib/index.js",
...
```

## Screenshots/Video
expected package content (from version 25.0.0):
<img width="223" alt="grafik" src="https://github.com/matrix-org/matrix-js-sdk/assets/32396733/db762563-2f94-4c39-b946-b7cc5ca84776">

package content (from version 25.1.0):
<img width="232" alt="grafik" src="https://github.com/matrix-org/matrix-js-sdk/assets/32396733/9a715c2f-46b2-4afa-a28e-1f2ab76b8910">

## Terminal output when running
```
node:internal/errors:490
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package '/Users/user/martix-test/node_modules/matrix-js-sdk/' imported from /Users/user/martix-test/src/index.js
    at new NodeError (node:internal/errors:399:5)
    at legacyMainResolve (node:internal/modules/esm/resolve:235:9)
    at packageResolve (node:internal/modules/esm/resolve:876:14)
    at moduleResolve (node:internal/modules/esm/resolve:938:20)
    at defaultResolve (node:internal/modules/esm/resolve:1153:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:838:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

```

## Quick fix
Downgrading npm package to `25.0.0`

## Tested on
- MacOS 13.3.1, node v18.15.0, npm 9.5.0
- MacOS 13.2.1, node v18.16.0, npm v9.5.1
- MacOS 13.2.1, node v18.16.0, yarn 1.22.4

---
cc @fg-uulm

**Comment by t3chguy** [comment]
This will be fixed in 25.1.1 tomorrow

**Comment by andybalaam** [comment]
Fixed in v25.1.1:

```
$ yarn init
...
Done in 4.90s.
$ yarn add matrix-js-sdk
...
Done in 5.31s.
$ ls node_modules/matrix-js-sdk/dist/
browser-matrix.js      browser-matrix.min.js
browser-matrix.js.map  browser-matrix.min.js.map
```

**Comment by andybalaam** [comment]
This was caused by a glitch in the release process for v25.1.0. We upgraded npm-publish to v2 which caused a problem because we were relying on behaviour from v1 which no longer worked. This caused our [npm publishing workflow](https://github.com/matrix-org/matrix-js-sdk/blob/develop/.github/workflows/release.yml) to fail. When we downgraded npm-publish and re-ran the workflow, some steps were skipped because they had already run before, so the package was incomplete.

Apologies for the inconvenience. In future if the npm publish fails and re-runs we will check the package more carefully.
